homeassistant/components/miele/climate.py

```diff
@@ -0,0 +1,244 @@
+            MieleAppliance.WINE_CABINET_FREEZER,
+        ),
+        description=MieleClimateDescription(
+            key="thermostat3",
+            value_fn=lambda value: cast(int, value.state_temperatures[2].temperature)
+            / 100.0,
+            target_fn=lambda value: cast(
+                int, value.state_target_temperature[2].temperature
+            )
+            / 100.0,
+            name="Zone 3",
+            zone=2,
+        ),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: MieleConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the climate platform."""
+    coordinator = config_entry.runtime_data
+
+    async_add_entities(
+        MieleClimate(coordinator, device_id, definition.description)
+        for device_id, device in coordinator.data.devices.items()
+        for definition in CLIMATE_TYPES
+        if (
+            device.device_type in definition.types
+            and (definition.description.value_fn(device) != -32768 / 100.0)
+        )
+    )
+
+
+class MieleClimate(MieleEntity, ClimateEntity):
+    """Representation of a climate entity."""
+
+    entity_description: MieleClimateDescription
+
+    def __init__(
+        self,
+        coordinator: MieleDataUpdateCoordinator,
+        device_id: str,
+        description: MieleClimateDescription,
+    ) -> None:
+        """Initialize the climate entity."""
+        super().__init__(coordinator, device_id, description)
+        self.api = coordinator.api
+        if (
+            self.device.device_type == MieleAppliance.FRIDGE_FREEZER
+            and self.entity_description.zone == 0
+        ):
+            name = DEVICE_TYPE_TAGS[MieleAppliance.FRIDGE]
+        elif (
+            self.device.device_type == MieleAppliance.FRIDGE_FREEZER
+            and self.entity_description.zone == 1
+        ):
+            name = DEVICE_TYPE_TAGS[MieleAppliance.FREEZER]
+        elif (
+            self.device.device_type == MieleAppliance.FRIDGE
+            and self.entity_description.zone == 0
+        ):
+            name = DEVICE_TYPE_TAGS[MieleAppliance.FRIDGE]
+        elif (
+            self.device.device_type == MieleAppliance.FREEZER
+            and self.entity_description.zone == 0
+        ):
+            name = DEVICE_TYPE_TAGS[MieleAppliance.FREEZER]
+        elif (
+            self.device.device_type == MieleAppliance.WINE_CABINET_FREEZER
+            and self.entity_description.zone == 0
+        ):
+            name = DEVICE_TYPE_TAGS[MieleAppliance.WINE_CABINET]
+        elif (
+            self.device.device_type == MieleAppliance.WINE_CABINET_FREEZER
+            and self.entity_description.zone == 1
+        ):
+            name = DEVICE_TYPE_TAGS[MieleAppliance.FREEZER]
+        else:
+            name = cast(str, self.entity_description.name)
+        self._attr_translation_key = name
+
+        zone = (
+            ""
```

Comment:
Why do we use an empty string? That leaves the unique id with `-` at the end, which is kinda meh imo